src/librustc_errors/lib.rs

```diff
@@ -575,13 +577,21 @@ impl Handler {
         (ret, diagnostics)
     }
 
+    pub fn code_emitted(&self, code: &DiagnosticId) -> bool {
```

Comment:
comment (short one would suffice)

e.g., `/// True if a diagnostic with this code has already been emitted. Used to suppress duplicate messages with -Zteach.`